pdfbox/src/main/java/org/apache/pdfbox/rendering/PageDrawer.java

```diff
@@ -382,25 +382,30 @@ else if (!(colorSpace instanceof PDPattern))
      */
     protected final void setClip()
     {
-        Area clippingPath = getGraphicsState().getCurrentClippingPath();
-        if (clippingPath != lastClip)
+        List<GeneralPath> clippingPaths = getGraphicsState().getCurrentClippingPaths();
+        if (clippingPaths != lastClips)
         {
-            if (clippingPath.getPathIterator(null).isDone())
-            {
-                // PDFBOX-4821: avoid bug with java printing that empty clipping path is ignored by
-                // replacing with empty rectangle, works because this is not an empty path
-                graphics.setClip(new Rectangle());
-            }
-            else
-            {
-                graphics.setClip(clippingPath);
-            }
+            transferClip(getGraphicsState(), graphics);
             if (initialClip != null)
             {
                 // apply the remembered initial clip, but transform it first
                 //TODO see PDFBOX-4583
             }
-            lastClip = clippingPath;
+            lastClips = clippingPaths;
+        }
+    }
+
+    protected void transferClip(PDGraphicsState graphicsState, Graphics2D graphics) {
+        Area clippingPath = graphicsState.getCurrentClippingPath();

```

Comment:
Subclasses can override this method and decide what to do by looking at `getCurrentClippingPaths()`. For instance, they could ignore the initial clipping path (page boundaries) if this seems reasonable.